src/mgr/PyModule.cc

```diff
@@ -174,19 +189,39 @@ int PyModule::load(PyThreadState *pMainThreadState)
       }
 
       // Configure sys.path to include mgr_module_path
-      std::string sys_path = std::string(Py_GetPath()) + ":" + get_site_packages()
-                             + ":" + g_conf->get_val<std::string>("mgr_module_path");
-      dout(10) << "Computed sys.path '" << sys_path << "'" << dendl;
-
+      string paths = (":" + get_site_packages() +
+		      ":" + g_conf->get_val<std::string>("mgr_module_path"));
+#if PY_MAJOR_VERSION >= 3
+      wstring sys_path(Py_GetPath() + wstring(begin(paths), end(paths)));
+      PySys_SetPath(const_cast<wchar_t*>(sys_path.c_str()));
+      dout(10) << "Computed sys.path '"
+	       << string(begin(sys_path), end(sys_path)) << "'" << dendl;
+#else
+      string sys_path(Py_GetPath() + paths);
       PySys_SetPath(const_cast<char*>(sys_path.c_str()));
+      dout(10) << "Computed sys.path '" << sys_path << "'" << dendl;
+#endif
     }
 
     PyMethodDef ModuleMethods[] = {
       {nullptr}
     };
+#if PY_MAJOR_VERSION >= 3
+    PyModuleDef ceph_module_def = {
+      PyModuleDef_HEAD_INIT,
+      "ceph_module",
+      nullptr,
+      -1,
+      ModuleMethods,
+    };
+#endif
 
     // Initialize module
+#if PY_MAJOR_VERSION >= 3
+    PyObject *ceph_module = PyModule_Create(&ceph_module_def);
```

Comment:
Why do we switch to PyModule_Create here but continue to use Py_InitModule for `ceph_logger` above?  Also it seems like the `ceph_logger_module` struct is unused, so I guess the intent was to migrate both calls?

Reply:
thanks @jcsp , the change related to `ceph_logger_module` was lost while rebasing against master. and i put it back in the latest change.